src/test/java/teammates/test/cases/ui/pagedata/InstructorFeedbackResultsPageDataTest.java

```diff
@@ -0,0 +1,1584 @@
+        
+        List<InstructorFeedbackResultsResponseRow> responseRows = questionPanel.getResponses();
+        assertEquals(5, responseRows.size());
+        // first response
+        InstructorFeedbackResultsResponseRow responseRow = responseRows.get(0);
+        verifyResponseRow(responseRow, dataBundle.feedbackResponses.get("response1ForQ1S1C1"), 
+                                       dataBundle.students.get("student1InCourse1"), 
+                                       dataBundle.students.get("student1InCourse1"));
+        
+        assertFalse(responseRow.isGiverProfilePictureAColumn());
+        assertFalse(responseRow.isRecipientProfilePictureAColumn());
+        assertNotNull(responseRow.getModerationButton());
+        
+        InstructorFeedbackResultsModerationButton modButton = responseRow.getModerationButton();
+  
+        verifyModerateButton(modButton, "Moderate Response", 
+                             dataBundle.students.get("student1InCourse1"), 
+                             "First feedback session", 1);
+        
+        //missing response
+        InstructorFeedbackResultsResponseRow missingResponseRow = responseRows.get(responseRows.size() - 1);
+        verifyResponseRow(missingResponseRow, null, dataBundle.students.get("student5InCourse1"), 
+                                             dataBundle.students.get("student5InCourse1"));
+        
+        assertNotNull(missingResponseRow.getModerationButton());
+        modButton = missingResponseRow.getModerationButton();
+        
+        String moderateButtonText = "Moderate Response";
+        StudentAttributes studentToModerate = dataBundle.students.get("student5InCourse1");
+        
+        int questionNumber = 1;
+        
+        verifyModerateButton(modButton, moderateButtonText, studentToModerate, feedbackSessionName,
+                                        questionNumber);
+        
+        // Verify the correctness of fourth question, which has giver type=instructor
+        //TODO see #3367
+        questionPanel = questionPanels.get(3);
+        responseRows = questionPanel.getResponses();
+        modButton = responseRows.get(0).getModerationButton();
+        //assertFalse(responseRows.get(0).isModerationButtonDisplayed());
+        //assertTrue(modButton.isAllowedToModerate());
+        
+        
+        ______TS("view by question, section 1, all questions");
+        
+        courseId = instructor.courseId;
+        feedbackSessionName = dataBundle.feedbackSessions.get("session1InCourse1").feedbackSessionName;
+        showStats = null;
+        groupByTeam = "on";
+        sortType = "question";
+        selectedSection = "Section 1";
+        data.setBundle(logic.getFeedbackSessionResultsForInstructorInSection(feedbackSessionName, courseId, 
+                                                                            instructor.email, selectedSection));
+        data.initForViewByQuestion(instructor, "Section 1", null, "on");
```

Comment:
Most of the time, you set `showStats` and `groupByTeam`, but don't use them in the init call, anyway.